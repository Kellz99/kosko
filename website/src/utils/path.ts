export function isAbsolute(path: string): boolean {
  return path.startsWith("/");
}

export function isRelative(path: string): boolean {
  return path.startsWith("./") || path.startsWith("../");
}

export function dirname(path: string): string {
  const index = path.lastIndexOf("/");
  if (index === -1) return path;
  return path.substring(0, index - 1) || "/";
}

export function extname(path: string): string {
  const index = path.lastIndexOf(".");
  if (index === -1) return "";
  return path.substring(index);
}

export function basename(path: string, ext?: string): string {
  const index = path.lastIndexOf("/");
  const base = index === -1 ? path : path.substring(index + 1);

  if (base.endsWith(ext)) {
    return base.substring(0, base.length - ext.length);
  }

  return base;
}

export function relative(from: string, to: string): string {
  const fromParts = from.split("/").filter(Boolean);
  const toParts = to.split("/").filter(Boolean);

  while (fromParts[0] && toParts[0]) {
    fromParts.shift();
    toParts.shift();
  }

  while (toParts[0] === "." || toParts[0] === "..") {
    const toPart = toParts.shift();

    if (toPart === "..") {
      fromParts.pop();
    }
  }

  while (fromParts.pop()) {
    toParts.unshift("..");
  }

  return toParts.join("/");
}
