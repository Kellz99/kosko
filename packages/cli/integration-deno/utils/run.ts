import execa from "execa";
import { join } from "node:path";
import { projectRoot } from "@kosko/test-utils";

export async function runDenoCLI(
  args: string[],
  options: execa.Options = {}
): Promise<execa.ExecaReturnValue> {
  return execa(
    "deno",
    [
      "run",
      "-A",
      "--import-map",
      join(projectRoot, "packages/cli/templates/deno/import_map.json"),
      "npm:kosko",
      ...args
    ],
    {
      ...options,
      env: {
        LC_ALL: "en_US",
        ...options.env
      }
    }
  );
}
