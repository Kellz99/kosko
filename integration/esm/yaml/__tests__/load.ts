import execa from "execa";
import { dirname, join } from "path";
import { installPackage } from "../../../run";

const testDir = dirname(__dirname);
let result: execa.ExecaReturnValue;

beforeAll(async () => {
  await installPackage(testDir, "yaml");
});

beforeEach(async () => {
  result = await execa.node(join(testDir, "index.mjs"), {
    cwd: testDir
  });
});

test("should load Kubernetes objects", () => {
  expect(result.stdout).toMatchSnapshot();
});
