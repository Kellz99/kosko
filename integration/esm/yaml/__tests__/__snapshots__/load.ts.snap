// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should load Kubernetes objects 1`] = `
"[
  IoK8sApiCoreV1Pod {
    apiVersion: 'v1',
    kind: 'Pod',
    metadata: { name: 'test-pod' }
  }
]"
`;
